### PR TITLE
Local llm

### PR DESCRIPTION
**Pull request by Sennaa**

### CODE REVIEW

File: ingest/ingester.py [hunk 1]
```diff
@@ -33,6 +34,7 @@ def __init__(self, collection_name: str, content_folder: str, vectordb_folder: s
         self.vecdb_type = settings.VECDB_TYPE if vecdb_type is None else vecdb_type
         self.chunk_size = settings.CHUNK_SIZE if chunk_size is None else chunk_size
         self.chunk_overlap = settings.CHUNK_OVERLAP if chunk_overlap is None else chunk_overlap
+        self.local_api_url = settings.LOCAL_API_URL if local_api_url is None and settings.LOCAL_API_URL is not None else local_api_url
```
Comment (StefanTroost) [inline ingest/ingester.py:37]: Hi @lrvl , @Sennaa , this parameter is still missing in settings.py
Shouldn't it be added there?

File: settings.py [hunk 2]
```diff
@@ -32,12 +33,15 @@
...
 LLM_MODEL_TYPE = "gpt35"
-# EMBEDDINGS_PROVIDER must be one of: "openai", "huggingface"
+# API_URL must be the URL to your (local) API
+# If LLM_TYPE is "local_llm" and model is run on your local machine, API_URL should be "localhost:11434" by default
+# API_URL = "localhost:11434"
+# EMBEDDINGS_PROVIDER must be one of: "openai", "huggingface", "local_embeddings"
 EMBEDDINGS_PROVIDER = "openai"
 # EMBEDDINGS_MODEL must be one of: "text-embedding-ada-002", "all-mpnet-base-v2"
 # If EMBEDDINGS_MODEL is "all-mpnet-base-v2" then EMBEDDINGS_PROVIDER must be "huggingface"
+# If EMBEDDINGS_MODEL is "local_embeddings" then EMBEDDINGS_PROVIDER must be one of the local models, e.g. "llama2"
```
Comment (StefanTroost) [inline settings.py:44]: Hi @Sennaa, @lrvl, I assume this line should be "If EMBEDDINGS_PROVIDER is "local_embeddings" then EMBEDDINGS_MODEL must be one of the local models e.g. "llama2" ?
Is "llama2" the LLM ánd embedding model you actually used for testing the user story?

Comment (lrvl) [reply settings.py:44]: Hi @StefanTroost . Yes, that's correct my current local settings.py uses EMBEDDINGS_MODEL = "llama2". During the user story we first tested with "orca-mini" and after that for a higher quality with "llama2".

I'd like to suggest a small change to how we handle settings.py in our GitHub repo in order to enhance security and personalization:

1. Rename `settings.py` to `settings_template.py` : This becomes a base template.
2. Add `settings.py` to `.gitignore`: Each developer creates their own settings.py from the template, which avoids sensitive data getting into the repo.

This practice should make our configuration management safer and flexible. Thoughts?

Comment (StefanTroost) [reply settings.py:44]: Hi @lrvl, that's a good idea indeed. I will first try to get the local LLM working together with the RAG pipeline on my Azure virtual machine (didn't succeed yesterday), and then do the changes regarding settings.py

**Review by StefanTroost — APPROVED** [review]

**Pull request merged**